pkg/block/fetcher.go

```diff
@@ -401,14 +401,26 @@ func NewLabelShardedMetaFilter(relabelConfig []*relabel.Config) *LabelShardedMet
 	return &LabelShardedMetaFilter{relabelConfig: relabelConfig}
 }
 
-// Filter filters out blocks that filters blocks that have no labels after relabelling.
+// Special block that will have ULID of the meta.json being referenced too.
+const blockIDLabel = "__block_id"
+
+// Filter filters out blocks that have no labels after relabelling of each block external (Thanos) labels.
 func (f *LabelShardedMetaFilter) Filter(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, _ bool) {
+	var lbls labels.Labels
 	for id, m := range metas {
-		if processedLabels := relabel.Process(labels.FromMap(m.Thanos.Labels), f.relabelConfig...); processedLabels != nil {
-			continue
+		lbls = lbls[:0]
+		if len(lbls) < len(m.Thanos.Labels)+1 {
```

Comment:
This check is now unnecessary. Here, `lbls` will ALWAYS be 0 length, and `len(m.Thanos.Labels)+1` will ALWAYS be at least 1, so the make will always be called. I think we can skip the `if` and simply always call the `lbls = make...`

Reply:
I wanted cap :man_facepalming:

Reply:
or just remove all of this and just reset lbls at the beginning of the for body, then append will take care of growing the array when needed

Reply:
Wanted to reuse the arr. Fixed now